### PR TITLE
Improve codegen types

### PR DESCRIPTION
**Pull request by deebov**
This PR improves the codegen types.

Thanks to these changes, we will be able reduce the size of the types package (i.e. `@ronin/acme`) because these improved types achieve the same result or even more without generating a lot of code.

For example, now you can also filter by properties of a nested related record:

```ts
await get.account.with.session.device('mac');
await get.account.with.session.device.notBeing('mac');
```

And you will be to access the properties of the nested related record (if it was included via `get.records.including(['nested_record']))`:

```ts
const account = await get.account({
    with: {
        handle: "leo",
    },
    including: ["session"],
});

console.log(account.session) // Session
```

Although this is not an ideal solution, this will be improved in the coming releases.

### CODE REVIEW

**Review by NuroDev — APPROVED** [review]

File: src/types/codegen.ts [hunk 1]
```diff
@@ -187,131 +204,144 @@ export namespace RONIN {
   type OrderedBy<TSchema, R, O> = OrderedByObject<TSchema, R, undefined, O> &
     OrderedByFunction<TSchema, R, O>;
 
+  // TODO: Remove this once its all references are removed
```
Comment (NuroDev) [inline src/types/codegen.ts:207]: References to what? `SchemaSlugKey`? And if so, can they be removed now or is this something that requires a new release for first?

File: src/types/codegen.ts [hunk 2]
```diff
@@ -187,131 +204,144 @@ export namespace RONIN {
...
     TSchema,
-    TReturn,
-    TSlug extends SchemaSlugKey,
     TVariant extends string = string,
     TOptions = undefined,
-    TWith = With<TSchema, TReturn, TOptions>,
+    TModifiedReturn = Records<Replace<TSchema, RONIN.RoninRecord, string>>,
   > extends ReducedFunction {
-    (
+    <TIncluding extends Including<TSchema> = []>(
```
Comment (NuroDev) [inline src/types/codegen.ts:254]: Amazing idea!!

File: src/types/codegen.ts (hunk 1 above)
Comment (deebov) [reply src/types/codegen.ts:207]: Ah, yes, the FRONT or RENDER might be using these that's why. I can add this to the comment.

Comment (deebov) [reply src/types/codegen.ts]: I just checked FRONT and RENDER, seems like this type is not being used anywhere. So I removed it.

**Pull request merged**